ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/adapter/ColorPickerAdapter.java

```diff
@@ -14,61 +15,94 @@
+        this.selectedPos = selectedPos;
     }
 
     @NonNull
     @Override
     public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
-        View view = inflater.inflate(R.layout.color_picker_item_list, parent, false);
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.color_picker_item_list, parent, false);
+
+        //Prepare the layout params in prior to avoid initializing it in onBindViewHolder
+        int height, width, margin = dpToPx(4);
+        height = width = dpToPx(50);
+        selectItemParams = new RelativeLayout.LayoutParams(width, height);
+        selectItemParams.setMargins(margin, margin, margin, margin);
+
+        height = width = dpToPx(40);
+        unselectItemParams = new RelativeLayout.LayoutParams(width, height);
+        unselectItemParams.setMargins(0, 0, 0, 0);
+
         return new ViewHolder(view);
     }
 
     @Override
     public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
         holder.colorPickerView.setBackgroundColor(colorPickerColors.get(position));
+        if (selectedPos == position) {
+            holder.border.setVisibility(View.VISIBLE);
+            holder.colorPickerView.setLayoutParams(selectItemParams);
+        } else {
+            holder.border.setVisibility(View.GONE);
+            holder.colorPickerView.setLayoutParams(unselectItemParams);
+        }
     }
 
     @Override
     public int getItemCount() {
         return colorPickerColors.size();
     }
 
-
     public void setOnColorPickerClickListener(OnColorPickerClickListener onColorPickerClickListener) {
         this.onColorPickerClickListener = onColorPickerClickListener;
     }
 
     class ViewHolder extends RecyclerView.ViewHolder {
-        View colorPickerView;
+        View colorPickerView, border;
 
         ViewHolder(View itemView) {
             super(itemView);
             colorPickerView = itemView.findViewById(R.id.color_picker_view);
+            border = itemView.findViewById(R.id.color_picker_background);
             itemView.setOnClickListener(v -> {
+                if (getAdapterPosition() == RecyclerView.NO_POSITION) return;
+
+                // Updating old as well as new positions
+                notifyItemChanged(selectedPos);
+                selectedPos = getAdapterPosition();
+                notifyItemChanged(selectedPos);
+
                 if (onColorPickerClickListener != null)
-                    onColorPickerClickListener.onColorPickerClickListener(colorPickerColors.get(getAdapterPosition()));
+                    onColorPickerClickListener.onColorPickerClickListener(
+                            colorPickerColors.get(getAdapterPosition()), selectedPos);
             });
         }
     }
 
     public interface OnColorPickerClickListener {
-        void onColorPickerClickListener(int colorCode);
+        void onColorPickerClickListener(int colorCode, int selectedColorPos);
```

Comment:
I only see one implementation that does not use the position.
added parameter not needed then?

Reply:
This is implemented in two places, `BrushConfigDialog.kt` and `TextEditorDialogFragment.java`. In  `BrushConfigDialog.kt` I need a position to retain the chosen color. But in case of `TextEditorDialogFragment.java` I ignore it.